src/axom/sidre/core/Group.cpp

```diff
@@ -1373,6 +1379,67 @@ Group* Group::deepCopyGroup(const Group* srcGroup, int allocID)
   return dstGroup;
 }
 
+/*
+ *************************************************************************
+ *
+ * Deep copy another Group into this Group.
+ *
+ * All contents in this group and their data are destroyed before the copy.
+ * The deep copy of a Group will copy the group hierarchy and deep copy
+ * all Views within the hierarchy.
+ *
+ *************************************************************************
+ */
+Group* Group::deepCopyGroupToSelf(const Group* srcGroup)
+{
+  SLIC_ERROR_IF(m_is_list && !srcGroup->m_is_list,
+                "Group::deepCopyToSelf cannot copy from a list Group '" + srcGroup->getPath() +
+                  "' to a non-list Group '" + getPath() + "'");
+
+  destroyGroupsAndData();
+  destroyViewsAndData();
+
+  // copy child Groups to new Group
+  for(auto& grp : srcGroup->groups())
+  {
+    deepCopyGroup(&grp);
+  }
+
+  // copy Views to new Group
+  for(auto& view : srcGroup->views())
+  {
+    deepCopyView(&view);
+  }
+
+  return this;
+}
+
+/*
+ *************************************************************************
+ *
+ * Reallocate data to View-specific allocators.
+ *
+ *************************************************************************
+ */
+Group* Group::reallocateTo(const std::function<int(const View&)>& viewToAllocatorId)
```

Comment:
Clarity:  

`reallocateTo` made me think it was going to move to a new group.
Would it be clearer if the function was just `reallocate()`?

Reply:
I've concluded that each `Group` needs to have 2 allocator ids, one for data usually involved in number crunching and one for metadata usually for managing the other data.  I've tried to do without this, but it's extremely messy and full of gotchas.  I gave up after several weeks of trying.  By allowing 2 allocator ids and using a backward-compatible extension of the current interface, we no longer need the `reallocateTo` function.